### DISCUSSION TITLE
Discussion about merged models

### DISCUSSION THREAD

**Discussion by buzzCraft** [EuroEval/EuroEval#209] [General]
Hi. First of all, thank you so much for setting up this framework.

I've seen that merged models have started to enter the leaderboard. While I'm excited by this, I'm also a bit concerned. Back in the summer of 2023, the [Hugging Face open_llm_leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard) was the place to go to keep up with the rapid development in the field, but my feeling is that nowadays it's just a leaderboard over who manages to pick the most test data contaminated models and merge them to get a bit higher on the ranking. If you look at some of the top models, they are often merges of merges of merges....

I know that it might be a bit ambitious, but I suggest that a new leaderboard is made for merged models, or at least introduce filters to be able to exclude them.

The one big drawback with this, of course, is that some might start merging and not report it as a merge.

What do you guys think?

**Comment by saattrupdan** [comment]
I completely agree with your worry, and I've been discussing it with several people now as well.

We're a group of researchers who's currently trying to investigate merges and how they work, and to combat the "infestation of the merges" to the official ScandEval leaderboard, we've created a [Hugging Face organisation](https://huggingface.co/merge-crew) that contains our merge experiments, as well as a private ScandEval leaderboard, where we only evaluate the models on validation splits. As such, only the final merged models will appear in the ScandEval leaderboard. 

That being said, there _are_ currently merged models in the leaderboard, so we need to figure out what to do with those. We could keep them in (and let those be the only merged models in there for a while), or we could remove all merged models for now, until our experiments are done (resulting in only 1-2 merged models to add to the leaderboard, if any).

How does that sound to you all?

**Comment by timpal0l** [comment]
I agree with both of you, I think we could add a "merged" tag like `"Contains a merge/moerge"`, inspired by [open_llm_leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard). But as long as one does not get any feedback from the test-set, I think its fine to put up merges, since in the end, it might be a good model - and the open source community can benefit from finding it on the ScandEval leaderboard, rather than not seeing it. Id say, keep it simple, with as few leaderboards as possible, but add more attributes to filter on. :)

**Comment by BirgerMoell** [comment]
I think everyone is making good points.

One thing that at least I am confused about is, how good are the merged models?

Do we have any evidence to see if they are performing worse than other models, if that is the case or even if we have a strong suspicion that so could be the case, then we need another method for eval or keep them separate.

Another way to think about it, is that previously you could complain that only people with GPUs can make contributions, now it's the other way around where anyone can contribute. If we are too against merged models, people could in theory make secret merge models and submit them, which seems less then ideal.

I think another solution is some form of human eval similar to the LLM arena, but someone needs to donate a GPU for that.

Maybe a toggle could be a good middle ground?

@buzzCraft you are super welcome to join us both on discord and on huggingface where we are trying to figure out how to best work with merge models.

https://huggingface.co/merge-crew

**Comment by saattrupdan** [comment]
> I agree with both of you, I think we could add a "merged" tag like `"Contains a merge/moerge"`, inspired by [open_llm_leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard). But as long as one does not get any feedback from the test-set, I think its fine to put up merges, since in the end, it might be a good model - and the open source community can benefit from finding it on the ScandEval leaderboard, rather than not seeing it. Id say, keep it simple, with as few leaderboards as possible, but add more attributes to filter on. :)

A toggle would maybe be a decent start, and the toggle should probably be off by default (i.e., not including m(o)erges), and people who are interested / open to using merges, can enable that toggle. Further, I think @timpal0l raises a good point about test set feedback. I could simply ensure that all merge contributions to the leaderboard has to be evaluated on the validation splits only, to avoid a nasty overfitting feedback loop.

**Comment by saattrupdan** [comment]
As a first step, I've now added an "Include merged models" checkbox to all leaderboards, which is off by default. When shown, these models have a different background colour to make the distinction clear. Further, all merged models *have* to be evaluated only on the validation split, otherwise it's not included in the leaderboard.

Any feedback to this would be appreciated!

**Comment by buzzCraft** [comment]
Sorry for not engaging in the discussion, didn't think it would generate a response so fast.

I think you found a good solution @saattrupdan, I even think it's fair to show merges by default, but that is just details.
I'll take you up on your invitation @BirgerMoell, I'm probably not in your server since the invite link (EDIT: Joined discord :) ) don't bring me anywhere, and to be honest, seeing the score of the merged models made me finally start the process that will lead to our next model. So thank you for that 😅
